File: Homework/Chau_Long/lesson_1/single_number.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        res = 0
        for num in nums:
            res ^= num
        return res

if __name__ == "__main__":
    sol = Solution()
    print(sol.singleNumber([2, 2, 1]) == 1)
    print(sol.singleNumber([4, 1, 2, 1, 2]) == 4)
    print(sol.singleNumber([1]) == 1)

File: Homework/Nguyen_Anh_Duc/single-number.go
package main

import "fmt"

func singleNumber(nums []int) int {
	var xor = 0
	for _, num := range nums {
		xor ^= num
	}
	return xor
}

func main() {
	fmt.Println(singleNumber([]int{2, 2, 1}))
	fmt.Println(singleNumber([]int{4, 1, 2, 1, 2}))
}
